test/python/test_docker_pack.py

```diff
@@ -0,0 +1,288 @@
+            containers = docker_client.containers.list(
+                all=True,
+                filters={'ancestor': image_name}
+            )
+
+            for c in containers:
+                c.remove(force=True)
+
+            # remove image itself
+            docker_client.images.remove(image_name)
+
+    request.addfinalizer(lambda: delete_image(image_name))
+    return {'name': image_name}
+
+
+# #####
+# Tests
+# #####
+def test_invalid_base_dockerfile(project_path, module_tmpdir, tmpdir):
+    invalid_dockerfile_path = os.path.join(tmpdir, 'Dockerfile')
+    with open(invalid_dockerfile_path, 'w') as f:
+        f.write('''
+            # Invalid dockerfile
+            FROM ubuntu:xenial
+        ''')
+
+    cmd = [
+        os.path.join(basepath, "cartridge"),
+        "pack", "docker",
+        "--from", invalid_dockerfile_path,
+        project_path,
+    ]
+    process = subprocess.run(cmd, cwd=module_tmpdir)
+    assert process.returncode == 1
+
+
+def test_docker_pack(project_path, docker_image, tmpdir, docker_client):
+    image_name = docker_image['name']
+    container = docker_client.containers.create(image_name)
+
+    container_distribution_dir = '/usr/share/tarantool/{}'.format(project_name)
+
+    # check if distribution dir was created
+    command = '[ -d "{}" ] && echo true || echo false'.format(container_distribution_dir)
+    output = run_command_on_image(docker_client, image_name, command)
+    assert output == 'true'
+
+    # get distribution dir contents
+    arhive_path = os.path.join(tmpdir, 'distribution_dir.tar')
+    with open(arhive_path, 'wb') as f:
+        bits, _ = container.get_archive(container_distribution_dir)
+        for chunk in bits:
+            f.write(chunk)
+
+    with tarfile.open(arhive_path) as arch:
+        arch.extractall(path=os.path.join(tmpdir, 'usr/share/tarantool'))
+    os.remove(arhive_path)
+
+    assert_dir_contents(
+        recursive_listdir(os.path.join(tmpdir, 'usr/share/tarantool/', project_name)),
+        skip_tarantool_binaries=True
+    )
+
+    assert_filemodes(tmpdir)
+    container.remove()
+
+    if tarantool_enterprise_is_used():
+        # check tarantool and tarantoolctl binaries
+        command = '[ -d "/usr/share/tarantool/tarantool-enterprise/" ] && echo true || echo false'
+        output = run_command_on_image(docker_client, image_name, command)
+        assert output == 'true'
+
+        command = 'cd /usr/share/tarantool/tarantool-enterprise/ && find .'
+        output = run_command_on_image(docker_client, image_name, command)
+
+        files_list = output.split('\n')
+        files_list.remove('.')
+
+        dir_contents = [
+            os.path.normpath(filename)
+            for filename in files_list
+        ]
+
+        assert 'tarantool' in dir_contents
+        assert 'tarantoolctl' in dir_contents
+    else:
+        # check if tarantool was installed
+        command = 'yum list installed 2>/dev/null | grep tarantool'
```

Comment:
I want to check if Tarantool was installed. If not, I will get an error on running `tarantool --version`,

Reply:
Yes, but "yum" is an implementation details. One day we will want to use tarantool installed from source and it will break tests

Reply:
If you get an error on running `tarantool --version` - tarantool is not installed

Reply:
If we start to install Tarantool from source, it will require a lot of changes as well as changing tests. As I now, building Tarantool from source isn't planned.

Reply:
You've already use 'tarantool --version' in `is_enterprise` function. It's an attempt to generalize the approach to versions management. I think it could be reused in several places

Reply:
`is_enterprise ` function checks Tarantool version on local machine, if Tarantool is not install there, test fill fail earlier.

Maybe it's better to use `tarantool --version` output instead of using `yum info tarantool` on version check.
But I don't want to change it now for one more reason - we are planning changes that allow to freeze Tarantool version in the result package or Docker image, so this check will be changed anyway (at least because Enterpise version would be tested too).